File: latihan 4 sekawan/KoleksiKendaraan.cpp
#include "KoleksiKendaraan.hpp"
#include <iostream>
using namespace std;

KoleksiKendaraan :: KoleksiKendaraan(){
    this->neff=0;
    this->capacity=100;
    this->storage = new Kendaraan[this->capacity];
}

KoleksiKendaraan :: KoleksiKendaraan(int size){
    this->neff=0;
    this->capacity=size;
    this->storage = new Kendaraan[this->capacity];
}

KoleksiKendaraan :: KoleksiKendaraan(const KoleksiKendaraan&koleksi){
    this->neff=koleksi.neff;
    this->capacity=koleksi.capacity;
    this->storage = new Kendaraan[this->capacity];
    for(int i=0;i<this->capacity;i++){
        this->storage[i] = koleksi.storage[i];
    }
}

KoleksiKendaraan :: ~KoleksiKendaraan(){
    delete [] this->storage;
}

void KoleksiKendaraan::PrintAll(){
    if(this->neff>=1){
        for(int i=0;i<this->neff;i++){
            this->storage[i].PrintInfo();
        }
    }
}

void KoleksiKendaraan::operator<<(Kendaraan &k ){
    if(this->neff < this->capacity){
        this->storage[this->neff] = k;
        this->neff++;
    }
}

void KoleksiKendaraan::operator<<(KoleksiKendaraan &k2){
    if (k2.neff>=1){
        for(int i=0;i<k2.neff;i++){
            this->operator<<(k2.storage[i]);
            this->neff++;
        }
    }
}

File: Latihan Bareng rava/KoleksiKendaraan.java
class KoleksiKendaraan {
    private int neff; // Neff array
    private int size; // Size array
    private Kendaraan[] Koleksi; // Array kendaraan

    // Default Konstruktor Koleksi Kendaraan
    public KoleksiKendaraan() {
        this.neff = 0; // Neff array
        this.size = 100; // Size array
        this.Koleksi = new Kendaraan[this.size]; // Array kendaraan
    }

    // Konstruktor Koleksi Kendaraan dengan parameter
    public KoleksiKendaraan(int size) {
        this.neff = 0; // Neff array
        this.size = size; // Size array
        this.Koleksi = new Kendaraan[this.size]; // Array kendaraan
    }

    // Method getKendaraanAt
    public Kendaraan getKendaraanAt(int id) {
        return this.Koleksi[id]; // Mengembalikan kendaraan pada index id
    }

    // Method addKendaraan
    public void add(Kendaraan obj_K) {
        this.Koleksi[this.neff] = obj_K; // Menambahkan kendaraan ke array
        this.neff++; // Menambahkan neff
    }

    // Method add in koleksiKendaraan
    public void add(KoleksiKendaraan obj_KK) {
        for (int i = 0; i < obj_KK.neff; i++) { // Looping sebanyak neff
            this.Koleksi[this.neff] = obj_KK.Koleksi[i]; // Menambahkan kendaraan ke array
            this.neff++; // Menambahkan neff
        }
    }

    // Method printAll
    public void printAll() {
        for (int i = 0; i < this.neff; i++) { // Looping sebanyak neff
            System.out.println("Kendaraan " + (i + 1)); // Print kendaraan
            this.Koleksi[i].printInfo(); // Print info kendaraan
            System.out.println(); // Print enter
        }
    }
}